libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -0,0 +1,366 @@
+                                   ",\"operation_results\":\"" + operation_result + "\",\"virtual_op\":" + virtual_op +
+                                   ",\"op\":\"" + op + "\"}";
+
+   // visitor data
+   std::string visitor_data = "";
+   if(_elasticsearch_visitor) {
+      operation_visitor o_v;
+      op_id(db).op.visit(o_v);
+
+      std::string fee_asset = fc::json::to_string(o_v.fee_asset);
+      std::string fee_amount = fc::json::to_string(o_v.fee_amount);
+      boost::replace_all(fee_amount, "\"", "");
+
+      std::string fee_data = "{\"fee_asset\":" + fee_asset + ", \"fee_amount\":\"" + fee_amount + "\"}";
+
+      std::string transfer_asset = fc::json::to_string(o_v.transfer_asset_id);
+      std::string transfer_amount = fc::json::to_string(o_v.transfer_amount);
+      boost::replace_all(transfer_amount, "\"", "");
+
+      std::string transfer_data = "{\"transfer_asset_id\":" + transfer_asset + ", \"transfer_amount\":\"" + transfer_amount + "\"}";
+      visitor_data = ",\"fee_data\": " + fee_data + ", \"transfer_data\": " + transfer_data;
+   }
+
+   // block data
+   auto block = db.fetch_block_by_number(op_id(db).block_num);
+   std::string block_num = std::to_string(block->block_num());
+   std::string block_time = block->timestamp.to_iso_string();
+   std::string trx_id = "";
+   // removing by segfault at blok 261319,- static variant wich thing.
+   //if(!block->transactions.empty() && block->transactions[op_id(db).trx_in_block].id().data_size() > 0 && block->block_num() != 261319 ) {
+   //   trx_id = block->transactions[op_id(db).trx_in_block].id().str();
+   //}
+   std::string block_data = "{\"block_num\":" + block_num + ",\"block_time\":\"" + block_time +
+                             "\",\"trx_id\":\"" + trx_id + "\"}";
+
+   // check if we are in replay or in sync and change number of bulk documents accordingly
+   uint32_t limit_documents = 0;
+   if((fc::time_point::now() - block->timestamp) < fc::seconds(30))
+      limit_documents = _elasticsearch_bulk_sync;
+   else
+      limit_documents = _elasticsearch_bulk_replay;
+
+   //wlog((account_transaction));
+   //wlog((operation_history));
+   //wlog((fee_data));
+   //wlog((transfer_data));
+   //wlog((block_data));
+
+   if(bulk.size() < limit_documents) { // we have everything, creating bulk array
+
+      // put alltogether in 1 line of bulk
+      std::string alltogether = "{\"account_history\": " + account_transaction + ", \"operation_history\": " + operation_history +
+                         ",\"operation_type\": " + op_type + ", \"block_data\": " + block_data + visitor_data + "}";
+
+      // bulk header before each line, op_type = create to avoid dups, index id will be ath id(2.9.X).
+      std::string _id = fc::json::to_string(ath.id);
+      bulk.push_back("{ \"index\" : { \"_index\" : \"graphene\", \"_type\" : \"data\", \"op_type\" : \"create\", \"_id\" : "+_id+" } }");
```

Comment:
Use the block time for generating a time-based index name, like graphene_YYYY-MM-DD.
https://www.elastic.co/guide/en/elasticsearch/guide/current/time-based.html#index-per-timeframe

Reply:
i like the approach but we will have to make an alias with all the indexes so we can make search globally. this is something we could do it from the plugin with the first op of the day. i don't feel very comfortable on having more than 1000 indexes, are you sure it will be the best ? what about sending the date in the `_type` field?

Reply:
You can search globally by specifying "graphene_*" as the index.
The time-based approach has a lot of advantages. It is correct that we should keep an eye on the number of indexes, perhaps use weekly or monthly instead of daily.
Log entries should have a timestamp anyway, there's no point in (ab)using the _type field for this.

Reply:
monthly sounds good to me, check https://github.com/bitshares/bitshares-core/pull/444/commits/4e7be5ccff85675a90351be1302d56f0f21ffc32

Reply:
sorry, added prefix `graphene-` so we can do the `*` search at https://github.com/bitshares/bitshares-core/pull/444/commits/430e75374a1b638600104564a30e03ba011d56b7